### PR TITLE
add extract

### PR DESCRIPTION
**Pull request by wuxiaobai24**
## What type of PR is this?
- [ ] bug
- [x] feature
- [ ] enhancement

#### Description:
Add a `extract` function to extract the substring that matches the regular expression from the string

## How do you solve it?
Use `std::regex`

## Checklist:
Tests:
- [x] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:
Add `extract` function
Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

File: src/common/function/test/FunctionManagerTest.cpp [hunk 1]
```diff
@@ -340,6 +340,14 @@ TEST_F(FunctionManagerTest, functionCall) {
                   std::vector<Value>({".", 1, true, Value::kNullValue, "world", time}),
                   "1.true.world.09:39:21.000012000");
   }
+  {
+    TEST_FUNCTION(extract, std::vector<Value>({"Hello Nebula Graph", "\\w+"}),
```
Comment (Shylock-Hg) [inline src/common/function/test/FunctionManagerTest.cpp:344]: Need no-ascii string.

File: src/common/function/FunctionManager.cpp [hunk 2]
```diff
@@ -417,6 +417,7 @@ std::unordered_map<std::string, std::vector<TypeSignature>> FunctionManager::typ
     {"duration",
      {TypeSignature({Value::Type::STRING}, Value::Type::DURATION),
       TypeSignature({Value::Type::MAP}, Value::Type::DURATION)}},
+    {"extract", {TypeSignature({Value::Type::STRING}, Value::Type::LIST)}},
```
Comment (Shylock-Hg) [inline src/common/function/FunctionManager.cpp:420]: (string, string) -> list

**Comment by Shylock-Hg** [comment]
Add test cases in tck.

**Review by yixinglu — APPROVED** [review]
LGTM

**Review by Shylock-Hg — APPROVED** [review]

**Comment by Sophie-Xie** [comment]
@wuxiaobai24 pls check the fail of make, thanks. https://github.com/vesoft-inc/nebula/runs/5966462671?check_suite_focus=true

**Pull request merged**